projects/packages/woocommerce-analytics/src/class-woo-analytics-trait.php

```diff
@@ -339,19 +341,33 @@ public function maybe_start_session() {
 			$session_id         = wp_generate_uuid4();
 			$this->session_id   = $session_id;
 			$this->landing_page = $this->get_current_url();
+			$session_expiration = $this->get_session_expiration_time();
 			$event_js           = $this->process_event_properties( 'woocommerceanalytics_session_started' );
 			$cookie_js          = "
             const sessionData = JSON.stringify({
                     session_id: '{$this->session_id}',
                     landing_page: encodeURIComponent('{$this->landing_page}'),
             });
-            document.cookie = `woocommerceanalytics_session=\${sessionData}; path=/; secure; samesite=strict`;
+            document.cookie = `woocommerceanalytics_session=\${sessionData}; expires={$session_expiration}; path=/; secure; samesite=strict`;
             ";
 			wc_enqueue_js( $cookie_js ); // save the session cookie for further events in the session
 			wc_enqueue_js( "_wca.push({$event_js});" ); // trigger session started event
 		}
 	}
 
+	/**
+	 * Get a 30 minutes expiration time from now or at midnight UTC, whichever comes first.
+	 *
+	 * @return string
+	 */
+	public function get_session_expiration_time() {
+		$thirty_minutes_from_now = time() + ( 30 * 60 ); // 30 minutes from now
+		$midnight                = strtotime( 'tomorrow UTC' ) - 1; // 1 second before midnight
+		$expiration_time         = min( $thirty_minutes_from_now, $midnight );
+		$date                    = new DateTime( "@$expiration_time", new DateTimeZone( 'UTC' ) );
+		return $date->format( 'D, d M Y H:i:s \G\M\T' );
```

Comment:
Since we already ensure the `$expiration_time` is in UTC, we can just using `return gmdate( 'D, d M Y H:i:s \G\M\T', $expiration_time );` so we can save a little bit performance by not initialising a DateTime object.

Reply:
Perfect @ianlin  Adjusted here https://github.com/Automattic/jetpack/pull/42286/commits/275e3c6c9740fdaeedc8249bec1c91b913431c1f